src/snapred/backend/data/LocalDataService.py

```diff
@@ -535,292 +340,155 @@ def writeNormalizationRecord(
         Persists a `NormalizationRecord` to either a new version folder, or overwrites a specific version.
         -- side effect: updates version numbers of incoming `NormalizationRecord` and its nested `Normalization`.
         """
-        runNumber = record.runNumber
-        stateId, _ = self._generateStateId(runNumber)
-        previousVersion = self._getLatestNormalizationVersionNumber(stateId, record.useLiteMode)
-        if version is None:
-            version = max(record.version, previousVersion + 1)
-        recordPath: Path = self.getNormalizationRecordFilePath(runNumber, record.useLiteMode, version)
-        record.version = version
-
-        # There seems no need to write the _nested_ Normalization,
-        # because it's written to a separate file during 'writeNormalizationState'.
-        # However, if it is going to be _nested_, this marks it with the correct version.
-        # (For example, use pydantic Field(exclude=True) to _stop_ nesting it.)
-        record.calibration.version = version
-
-        normalizationPath: Path = self._constructNormalizationDataPath(runNumber, record.useLiteMode, version)
-        # check if directory exists for runId
-        if not normalizationPath.exists():
-            os.makedirs(normalizationPath)
-        # append to record and write to file
-        write_model_pretty(record, recordPath)
+
+        indexor = self.normalizationIndexor(record.runNumber, record.useLiteMode)
+        # write the record to file
+        indexor.writeRecord(record, version)
+        # separately write the normalization state
+        indexor.writeParameters(record.calculationParameters, version)
+
         logger.info(f"wrote NormalizationRecord: version: {version}")
         return record
 
-    def writeNormalizationWorkspaces(self, record: NormalizationRecord) -> NormalizationRecord:
+    def writeNormalizationWorkspaces(self, record: NormalizationRecord, version: Optional[int] = None):
         """
         Writes the workspaces associated with a `NormalizationRecord` to disk:
         -- assumes that `writeNormalizationRecord` has already been called, and that the version folder exists
         """
-        normalizationDataPath: Path = self._constructNormalizationDataPath(
-            record.runNumber, record.useLiteMode, record.version
-        )
+        indexor = self.normalizationIndexor(record.runNumber, record.useLiteMode)
+        version = indexor.thisOrNextVersion(version)
+        normalizationDataPath: Path = indexor.versionPath(version)
         for workspace in record.workspaceNames:
-            filename = workspace + "_" + wnvf.formatVersion(record.version)
+            filename = workspace + "_" + wnvf.formatVersion(version)
```

Comment:
All of this crap needs to be moved _back_ into the "save" method that calls this method.